### PR TITLE
Fork this plugin and customize it

### PR DESCRIPTION
**Pull request by mrsimo**
/cc @stankevich 

I ended up realizing that the new plugin I implemented was very similar to the original one, only I wanted a slightly different format. So I've forked the original instead, removed some things we don't need and tweaked others.

My idea at the moment would be to just merge this, tag some version for release, and never touch this repo ever again :smile_cat:

### CODE REVIEW

File: hooks/command [hunk 1]
```diff
@@ -30,14 +29,17 @@ buildkite-agent artifact download \
 
 echo "--- :junit: Processing the junits"
 
+# Move ruby code to current dir so we can mount it
+plugin_ruby_dir="$(cd "$( dirname "${BASH_SOURCE[0]}" )/../ruby" && pwd)"
+ruby_dir="$(pwd)/$(mktemp -d "junit-annotate-plugin-ruby-tmp.XXXXXXXXXX")"
+cp -R "${plugin_ruby_dir}" "${ruby_dir}"
```
Comment (mrsimo) [inline hooks/command]: I guess you already know, but the original was trying to mount something in `/buildkite/plugins`, which is not [in the list of volumes we mount](https://github.com/harvesthq/ci-cd/blob/978ccda9673a22ede9edafb5a03cae15de4c7ebc/.kubernetes/manifests/buildkite-agent/deployment-buildkite-agent.yml.erb#L45-L55). My solution was to instead copy the ruby files to a folder inside the current build's path.

Do you like this?

File: ruby/bin/annotate [hunk 2]
```diff
@@ -44,58 +31,66 @@ def message_content(element)
...
-  if failure.message
-    puts "<p>#{failure.message.chomp.strip}</p>\n\n"
-  end
+  puts "<summary><code>#{failure.suite_name}##{failure.name}</code></summary>\n\n"
   if failure.body
-    puts "<pre><code>#{CGI.escapeHTML(failure.body.chomp.strip)}</code></pre>\n\n"
-  end
-  if failure.job
-    puts "in <a href=\"##{failure.job}\">Job ##{failure.job}</a>"
+    puts "<pre class=\"term\"><code>#{CGI.escapeHTML(failure.message.chomp.strip)}\n\n#{CGI.escapeHTML(failure.body.chomp.strip)}</code></pre>\n\n"
```
Comment (mrsimo) [inline ruby/bin/annotate:93]: This is what these changes make it look like: https://buildkite.com/harvesthq/harvest-id/builds/1015

File: hooks/command (hunk 1 above)
Comment (stankevich) [reply hooks/command]: Sure! Although it looks like you only need the `annotate` script from the `ruby/` dir, right? If that's the case, I'd only copy that file and tweak the `command` script a bit:

```bash
annotate_src="${BASH_SOURCE%/*}/../ruby/bin/annotate"
annotate_path="${PWD}/.buildkite/bin/annotate-tests"
cp -a "${annotate_src}" "${annotate_path}"

docker \
  --log-level "error" \
  run \
  --rm \
  --volume "${annotate_path}:/usr/local/bin/anotate-tests" \
  --volume "${artifacts_dir}:/junits" \
  ruby:2.7-alpine ruby /usr/local/bin/anotate-tests /junits \
    > "$annotation_path"
```

I'm fine with mounting the `plugins` dir across Buildkite and Docker-in-Dockers too. Whatever's easier!

**Review by stankevich — APPROVED** [review]
Just a minor non-blocking suggestion. Looks fine as is!

**Comment by mrsimo** [comment]
Thanks a lot, as always, @stankevich. I thought about mounting the plugins dir as well... but whatever, since we're foking anyway, might as well fix it this way. Let's keep it in mind for future usage!

**Pull request merged**